test/test_appscale_reset_pwd.py

```diff
@@ -138,18 +133,10 @@ def test_reset_password_for_user_that_doesnt_exist(self):
       .and_return('nothing interesting here') \
       .and_return('Database is at not-up-yet') \
       .and_return('Database is at public1')
-
     flexmock(SOAPpy)
     SOAPpy.should_receive('SOAPProxy').with_args('https://public1:17443') \
       .and_return(fake_appcontroller)
 
-    # and mock out changing the password
-    fake_userappserver = flexmock(name='fake_userappserver')
```

Comment:
So when a user doesn't exist, isn't the reset_password method called?
If it is then it should be mocked here like it is for the test above.